Src/Newtonsoft.Json/JsonTextWriter.cs

```diff
@@ -817,20 +817,77 @@ private void WriteIntegerValue(long value)
             }
         }
 
-        private void WriteIntegerValue(ulong uvalue, bool negative)
+        private void WriteIntegerValue(ulong value, bool negative)
         {
-            if (!negative & uvalue <= 9)
+            if (!negative & value <= 9)
             {
-                _writer.Write((char)('0' + uvalue));
+                _writer.Write((char)('0' + value));
             }
             else
             {
-                int length = WriteNumberToBuffer(uvalue, negative);
+                int length = WriteNumberToBuffer(value, negative);
                 _writer.Write(_writeBuffer, 0, length);
             }
         }
 
         private int WriteNumberToBuffer(ulong value, bool negative)
+        {
+            if (value < uint.MaxValue)
```

Comment:
Should this be `<` or `<=`?

It would be a good idea to have unit tests that test around this boundary (just smaller than MaxValue, equals MaxValue, just greater than MaxValue).